pkg/parse/state.go

```diff
@@ -107,7 +107,11 @@ func (s *reconcilerState) reset() {
 // invalidate logs the errors, clears the state tracking information.
 // invalidate does not clean up the `s.cache`.
 func (s *reconcilerState) invalidate(errs status.MultiError) {
-	klog.Errorf("Invalidating reconciler checkpoint: %v", status.FormatSingleLine(errs))
+	if status.HasTransientErrors(errs) {
```

Comment:
Will there ever be some transient and some non-transient?

Reply:
It is possible.

The `invalidate` function is called after `setSourceStatus`: https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/parse/run.go#L164.

If `setSourceStatus` has errors, it will be appended to the transient error.